jwst/ramp_fitting/ramp_fit.py

```diff
@@ -303,92 +328,237 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
+    #   noise only, read noise only, and the combination of Poisson noise and
+    #   read noise. The integration-specific variances are 3D arrays, and
+    #   the segment-specific variances are 4D arrays . The naming convention for
+    #   the arrays:
+    #     'var': a variance
+    #     'p3': intermediate 3D array for variance due to Poisson noise
+    #     'r4': intermediate 4D array for variance due to read noise
+    #     'both4': intermediate 4D array for ocmbined variance due to both
+    #               Poisson and read noise
+    #     'inv_<X>': intermediate array = 1/<X>
+    #     's_inv_<X>': intermediate array = 1/<X>, summed over integrations
+
+    # Loop over data integrations
+    for num_int in range(n_int):
+
+        # loop over data sections
+        for rlo in range(0, cubeshape[1], nrows):
+            rhi = rlo + nrows
+
+            if rhi > cubeshape[1]:
+                rhi = cubeshape[1]
+
+            # data_sect and gdq_sect are: [ groups, y, x ]
+            data_sect = model.get_section('data')[num_int, :, rlo:rhi, :]
+            gdq_sect = gdq_cube[num_int, :, rlo:rhi, :]
+
+            rn_sect = readnoise_2d[rlo:rhi, :]
+            gain_sect = gain_2d[rlo:rhi, :]
+
+            #  Calculate results needed to compute the variance arrays
+            den_r3, den_p3, num_r3, segs_beg_3 = \
+                utils.calc_slope_vars( rn_sect, gain_sect, gdq_sect,
+                                       group_time, max_seg )
+
+            segs_4[num_int, :, rlo:rhi, :] = segs_beg_3
+            var_p4[num_int, :, rlo:rhi, :] = den_p3 * med_slopes[ rlo:rhi, :]
+            var_r4[num_int, :, rlo:rhi, :] = num_r3 * den_r3
+
+        var_p4[var_p4 < 0.] = 0.
+        var_p4 = utils.rem_nan_inf(var_p4)
+
+        var_r4[var_r4 < 0.] = 0.
+        var_r4 = utils.rem_nan_inf(var_r4)
+
+        inv_var_p4[num_int, :, :, :] = 1./var_p4[num_int, :, :, :]
+        inv_var_p4 = utils.rem_nan_inf(inv_var_p4)
+        s_inv_var_p3[num_int, :, :] = (inv_var_p4[num_int, :, :, :]).sum(axis=0)
 
-        opt_res.ped_int[num_int, :, :] = \
+        var_p3[num_int, :, :] = 1./ s_inv_var_p3[num_int, :, :]
+        var_p3 = utils.rem_nan_inf(var_p3)
+
+        inv_var_r4[num_int, :, :, :] = 1./var_r4[num_int, :, :, :]
+        inv_var_r4 = utils.rem_nan_inf(inv_var_r4)
+ 
+        s_inv_var_r3[num_int, :, :] = (inv_var_r4[num_int, :, :, :]).sum(axis=0)
+        var_r3[num_int, :, :] = 1./ s_inv_var_r3[num_int, :, :]
+        var_r3 = utils.rem_nan_inf(var_r3)
+
+        var_both4[ num_int,:,:,:] = var_r4[num_int,:,:,:] + var_p4[num_int,:,:,:]
+        inv_var_both4[num_int, :, :, :] = 1./var_both4[num_int, :, :, :]
+        inv_var_both4 = utils.rem_nan_inf(inv_var_both4)
+
+        # Want to retain values in the 4D arrays only for the segments that each
+        #   pixel has, so will zero out values for the higher indices. Creating
+        #   and manipulating intermediate arrays (views, such as var_p4_int)
```

Comment:
It looks as if the close parenthesis should be moved to the next line.

Reply:
Right, updated.